src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -508,14 +508,24 @@ public void serializeToView(@NotNull final ByteBuf buf)
             buf.writeInt(data == null ? 0 : data.getId());
         }
 
-        buf.writeInt(recipes.size());
+        final List<IRecipeStorage> storages = new ArrayList<>();
         for(final IToken token: new ArrayList<>(recipes))
         {
-            if(ColonyManager.getRecipeManager().getRecipes().get(token) == null)
+            final IRecipeStorage storage = ColonyManager.getRecipeManager().getRecipes().get(token);
+            if(storage == null)
             {
                 removeRecipe(token);
             }
-            ByteBufUtils.writeTag(buf, StandardFactoryController.getInstance().serialize(token));
+            else
+            {
+                storages.add(storage);
+            }
+        }
+
+        buf.writeInt(storages.size());
```

Comment:
Why are we writing the entire recipe? Are we not supposed to use the central storage?

Reply:
I also thought that, but else we'd have to send over 5000 recipes on login and then a new one each time one changes.